src/test/DelegationFaucet.t.sol

```diff
@@ -0,0 +1,219 @@
+            stratBalanceAfter,
+            stratBalanceBefore + _depositAmount,
+            "test_mintDepositAndDelegate_TokenBalances: strategy balance not updated correctly"
+        );
+        assertEq(
+            stakeToken.balanceOf(stakerContract),
+            0,
+            "test_mintDepositAndDelegate_TokenBalances: staker balance should be 0"
+        );
+    }
+
+    /**
+     * @notice Check the before/after values for strategy shares and operator shares
+     * @param _operatorIndex is the index of the operator to use from the test-data/operators.json file
+     * @param _depositAmount is the amount of stakeToken to mint to the staker and deposit into the strategy
+     */
+    function test_mintDepositAndDelegate_StrategyAndOperatorShares(
+        uint8 _operatorIndex,
+        uint256 _depositAmount
+    ) public {
+        cheats.assume(_operatorIndex < 15 && _depositAmount < DEFAULT_AMOUNT);
+        if (_depositAmount == 0) {
+            _depositAmount = DEFAULT_AMOUNT;
+        }
+        // Setup Operator
+        address operator = getOperatorAddress(_operatorIndex);
+        address stakerContract = delegationFaucet.getStaker(operator);
+        _registerOperator(operator);
+
+        // Mint token to Staker, deposit minted amount into strategy, and delegate to operator
+        uint256 stakerSharesBefore = strategyManager.stakerStrategyShares(stakerContract, stakeTokenStrat);
+        uint256 operatorSharesBefore = delegation.operatorShares(operator, stakeTokenStrat);
+        delegationFaucet.mintDepositAndDelegate(operator, _depositAmount);
+
+        uint256 stakerSharesAfter = strategyManager.stakerStrategyShares(stakerContract, stakeTokenStrat);
+        uint256 operatorSharesAfter = delegation.operatorShares(operator, stakeTokenStrat);
+        assertTrue(
+            delegation.delegatedTo(stakerContract) == operator,
+            "test_mintDepositAndDelegate_StrategyAndOperatorShares: delegated address not set appropriately"
+        );
+        assertTrue(
+            delegation.isDelegated(stakerContract),
+            "test_mintDepositAndDelegate_StrategyAndOperatorShares: delegated status not set appropriately"
+        );
+        assertEq(
+            stakerSharesAfter,
+            stakerSharesBefore + _depositAmount,
+            "test_mintDepositAndDelegate_StrategyAndOperatorShares: staker shares not updated correctly"
+        );
+        assertEq(
+            operatorSharesAfter,
+            operatorSharesBefore + _depositAmount,
+            "test_mintDepositAndDelegate_StrategyAndOperatorShares: operator shares not updated correctly"
+        );
+    }
+
+    /**
+     * @notice Invariant test the before/after values for strategy shares and operator shares from multiple runs
+     */
+    /// forge-config: default.invariant.runs = 5
+    /// forge-config: default.invariant.depth = 20
+    function invariant_test_mintDepositAndDelegate_StrategyAndOperatorShares() public {
```

Comment:
just so I understand correctly, what exactly is this being labeled an invariant test doing?
seems like its set to do 5 runs of 20 calls, but like, what's being done in each run? I would think you define something like a setup to be done once and a bunch of conditions to check after each call, but this doesn't quite look like that -- does this run all of the below, do a random function call, then another run through all of the below + another random function call, etc.?

Reply:
You can set targetContracts that will be called over course of a invariant test run. It defaults to all contracts defined in setUp. In each of those 5 runs, it will call 20 functions randomly with fuzzed inputs and then check your invariant function and make sure the assertions are still true.

Reply:
Gotcha, makes sense.
So it's really just checking that every time `mintDepositAndDelegate` is called, the share amounts all change by the right amount and the staker contract is properly delegated, right? And the random calls here are kind of seeing if you can get into some state which messes up that behavior?
LGTM.